addons/analytic/models/analytic_plan.py

```diff
@@ -233,7 +233,7 @@ def _get_applicability(self, **kwargs):
         else:
             score = 0
             applicability = self.default_applicability
-            for applicability_rule in self.applicability_ids:
+            for applicability_rule in self.applicability_ids.filtered(lambda rule: rule.company_id == self.env.company):
```

Comment:
I suppose this shouldn't be checked on the company in self.env, but rather on the company set on the object we're willing to apply some analytic plan (I can encode a invoice for a company while my current company is another one). 

So, maybe checking on the company from the context first? :thinking:

Reply:
Thanks for the suggestion, I've updated the solution to retrieve the company from context data

Reply:
Indeed it's better now, thanks!